lib/AST/Decl.cpp

```diff
@@ -5335,6 +5335,9 @@ bool AbstractFunctionDecl::isObjCInstanceMethod() const {
 }
 
 static bool requiresNewVTableEntry(const AbstractFunctionDecl *decl) {
+  if (!isa<ClassDecl>(decl->getDeclContext()))
```

Comment:
Do protocols not use this too?

Does this get called on non-class members at all?

Reply:
Is it possible to make this an assertion instead, then? Or would that mess up call sites too much?

Reply:
It messed up a bunch of call sites. I'm not even sure why I included this patch here, it was just something I noticed while looking at the code.